distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/RestitoRetrievalResponse.java

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.common.test.restito;
+
+import static java.lang.Thread.sleep;
+
+import static com.xebialabs.restito.semantics.Action.composite;
+import static com.xebialabs.restito.semantics.Action.contentType;
+import static com.xebialabs.restito.semantics.Action.custom;
+import static com.xebialabs.restito.semantics.Action.header;
+
+import java.io.IOException;
+
+import org.glassfish.grizzly.http.server.Response;
+import org.slf4j.LoggerFactory;
+import org.slf4j.ext.XLogger;
+
+import com.xebialabs.restito.semantics.Action;
+import com.xebialabs.restito.semantics.Function;
+
+/**
+ * Encompasses the response for a product retrieval for a Restito stub server.
+ * Allows for the simple setup of specific responses, slow sending, and simulated
+ * network failure retry testing.
+ */
+public class RestitoRetrievalResponse implements Function<Response, Response> {
```

Comment:
How about `ChunkedContent` instead? That would be in line with the other Restito actions such as `resourceContent` and `stringContent`.

I was also thinking, maybe we should make the constructors `private` and have static factory methods like `Action chunkedContent(String responseMessage)`, `Action chunkedContent(String responseMessage, int messageDelayMs)`, etc., to replace the current `action()` method. With static imports, people would be able to use this like any of the other Restito Actions. Thoughts?